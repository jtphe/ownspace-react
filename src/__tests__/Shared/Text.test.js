import React from 'react';
import OSText from '@shared/Text';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import { Text } from 'react-native';
import Adapter from 'enzyme-adapter-react-16';

describe('Text tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OSText />);
  });

  it('Text snapshot', async () => {
    let component;

    await renderer.act(async () => {
      component = renderer.create(<OSText />);
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Check OwnSpace's font", () => {
    const osText = wrapper.find(Text).get(0);
    expect(osText.props.style[1].fontFamily).toEqual('HelveticaNeue');
  });

  it('Text selectable', () => {
    const osText = wrapper.find(Text).get(0);
    expect(osText.props.selectable).toEqual(true);
  });
});

Enzyme.configure({ adapter: new Adapter() });
