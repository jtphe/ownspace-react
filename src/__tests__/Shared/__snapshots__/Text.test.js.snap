// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text tests Text snapshot 1`] = `
<Text
  ellipsizeMode="tail"
  selectable={true}
  style={
    Array [
      null,
      Object {
        "fontFamily": "HelveticaNeue",
      },
    ]
  }
/>
`;
